csharp/src/Ice/WSTransceiver.cs

```diff
@@ -13,6 +13,23 @@
 
 namespace IceInternal
 {
+
+    // TODO: refactor and eliminate these exceptions
+    public sealed class WSProtocolException : Exception
```

Comment:
Sounds like we could also use InvalidDataException here instead?

Reply:
All these WS exceptions are thrown and caught in this file. I don't think reusing other exceptions is correct. Ideally, we would not use exceptions for this status communication.